src/spider/worker/ExecutorHandle.hpp

```diff
@@ -0,0 +1,33 @@
+#ifndef SPIDER_WORKER_EXECUTORHANDLE_HPP
+#define SPIDER_WORKER_EXECUTORHANDLE_HPP
+
+#include <mutex>
+#include <optional>
+
+#include <boost/uuid/uuid.hpp>
+
+#include "TaskExecutor.hpp"
+
+namespace spider::worker {
+/**
+ * This class acts as a handle for thread-safe access to the task executor and task id.
```

Comment:
If I understand correctly, a TaskExecutor will only ever own a single task (i.e. I don't think it is possible to re-use a TaskExecutor for a new task or for it to switch tasks). If I'm correct, we can just store the task ID when we create the TaskExecutor to avoid extra working keeping them in sync.